src/ui/port-selector/js/port-select-box.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Flocking UI Select Box
+ *   Copyright 2014-2019, Colin Clark and Tony Atkins
+ *
+ * Dual licensed under the MIT and GPL Version 2 licenses.
+ */
+
+"use strict";
+
+var flock = fluid.registerNamespace("flock");
+
+// TODO: Generalize this and merge into Flocking's
+// flock.ui.selectBox component.
+fluid.defaults("flock.midi.portSelectBox", {
+    gradeNames: ["flock.ui.selectBox"],
+
+    preferredPort: undefined,
+
+    listeners: {
+        "onRender.selectInitial": {
+            priority: "after:renderOptions",
+            funcName: "flock.midi.portSelectBox.selectInitial",
+            args: "{that}"
+        }
+    }
+});
+
+flock.midi.portSelectBox.selectInitial = function (that) {
+    var optionsLength = fluid.get(that, "model.options.length");
+    if (optionsLength) {
+        if (!that.model.selection && that.options.preferredPort) {
```

Comment:
This check ends up being called after preferredPort has been remapped to preferredOption, which I assume is a convention from the select box stuff?  Anyway, as far as I can tell that's why the port affinity doesn't work at the moment.

Reply:
Thanks, that's a typo. I had thought about trying to generalize this "preferred selection" (or whatever we'd call it) logic and move it to the Flocking select box, but realized I was fully clear on how it was implemented and that I should wait. I'll correct this.

Reply:
Thanks, that fixed the device affinity.